### DISCUSSION TITLE
How do you export an STL file in pyfluent?

### DISCUSSION THREAD

**Discussion by forharrisonbaby** [ansys/pyfluent#3411] [Q&A]
This is the main section of code where I try to export a mesh file:
```
generate_volume_mesh = workflow.TaskObject["Generate the Volume Mesh"]
generate_volume_mesh.Arguments.update_dict({"VolumeFill": "poly-hexcore"})
generate_volume_mesh.Execute()
 
mesh_file = r"C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI\my_mesh.stl"
session.tui.file.export.stl("ANSYS Fluent", "2024R1", "meshing", mesh_file)
```
The error happens at the last line of code. When that code is ran, it tells me this:
```
Exception has occurred: RuntimeError
menu not found
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
    status = StatusCode.UNKNOWN
    details = "menu not found"
    debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2024-10-20T20:51:02.9016385+00:00", grpc_status:2, grpc_message:"menu not found"}"
>
 
During handling of the above exception, another exception occurred:
 
  File "C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI\fluentTest2.py", line 109, in <module>
    session.tui.file.export.stl("ANSYS Fluent", "2024R1", "meshing", mesh_file)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: menu not found
```
In the Pyfluent documentation, it says that the stl class has 4 inputs; service, version, mode and path. It's possible that these inputs are wrong, but I'm not really sure what they should be changed to. It's also possible that the function is wrong, but again I'm not sure. I also tried this:
`session.tui.file.export.stl(file_name = r"C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI\my_mesh.stl")`
But it basically gave what looks like the exact same error:
```
Exception has occurred: RuntimeError
menu not found
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "menu not found"
	debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2024-10-21T17:58:41.4064292+00:00", grpc_status:2, grpc_message:"menu not found"}"
>

During handling of the above exception, another exception occurred:

  File "C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI\fluentTest2.py", line 203, in <module>
    session.tui.file.export.stl(file_name = r"C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI\my_mesh.stl")
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: menu not found
```
I don't know if I have to make the file beforehand or what, but nothing I've tried allows me to export the stl.

**Comment by ekostson** [comment]
@mkundu1 , @seanpearsonuk 

are you able to assist here with this question ?

**Comment by hpohekar** [comment]
@forharrisonbaby @seanpearsonuk @mkundu1 @ekostson 

Please use it as follows:

`meshing.tui.file.export.stl("D:\pyfluent\FM_AAPQ13pUvWxQtxN_86052\TaskObject23.msh.h5")`

![image](https://github.com/user-attachments/assets/bd2fd45c-ff84-4b0b-bf7d-f5c4a1a834ba)

You can refer the TUI command usage [here](https://fluent.docs.pyansys.com/version/stable/user_guide/legacy/tui.html).

**Reply by ekostson** [reply]
Many thanks @hpohekar 

Just one thing that is confusing, why can we not add the .stl format (you are saving as msh,h5) format?

So why not:

meshing.tui.file.export.stl("D:\pyfluent\FM_AAPQ13pUvWxQtxN_86052\TaskObject23.stl") ?

**Reply by forharrisonbaby** [reply]
@hpohekar @ekostson 
Sorry for the late response, but this also didn't work. At first I tried your code exactly, but I just got the same error. Since my computer doesn't have that exact same file path, I wrote code to capture the newly generated FM file and put my file in there...

```
def get_latest_folder(parent_directory):
    subdirs = [os.path.join(parent_directory, d) for d in os.listdir(parent_directory) if os.path.isdir(os.path.join(parent_directory, d))]
    latest_folder = max(subdirs, key=os.path.getmtime)
    return latest_folder + "\my_mesh.msh.h5"

latest_folder = get_latest_folder(r"C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI") 
print(f"Latest folder: {latest_folder}")
session.tui.file.export.stl(latest_folder)
```

But still nothing. Eventually I tried using the help command, and this really started to raise some eyebrows for me. Firstly, I did the help command on `help(session.tui.file.export)`. It worked as expected, and it even told me of the stl class.

```
 stl = <class 'meshing_tui_241.main_menu.file.export.stl'>
     Write a STL boundary mesh file.
```

But right when I ran the help command on the infamous session.tui.file.export.stl with `help(session.tui.file.export.stl)`, boom.

```
Exception has occurred: RuntimeError
menu not found
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "menu not found"
	debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"menu not found", grpc_status:2, created_time:"2024-11-29T02:33:04.981497+00:00"}"
>

During handling of the above exception, another exception occurred:

  File "C:\Users\forha\OneDrive\Documents\GitHub\F1car-wAI\fluentTest2.py", line 217, in <module>
    help(session.tui.file.export.stl)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: menu not found
```

Now I'm thinking it might not be a file problem after all. It might be a version issue, or something else because i think it's saying that the stl export menu may not be available in the mode I'm running, and the help command isn't something that needs a file. I've linked my entire code below, and this section starts goes through lines 117-134. I have the most recent version of ansys.fluent.core, 0.28.dev0.

[fluentTest.txt](https://github.com/user-attachments/files/17953992/fluentTest.txt)

**Reply by forharrisonbaby** [reply]
Or maybe could you share your code so I could compare mine to yours?
